### ISSUE TITLE
Guidelines for marking external machines used as part of our CI as required

### ISSUE THREAD

**Issue by fidencio** [kata-containers/kata-containers#6936]
As we're moving to use GitHub actions, and at the same time bringing the `CCv0` branch into the `main` branch, we've started using external machines as part of the CI.

Right now we have 3 machines, all used for the Confidential Containers work, and at some point we'd like to mark them as required as part of our CI.  However, we need to have a guideline about that, as it'd be nice to avoid blocking the community's progress because of breakages on CIs that the community does *not* easily have access to.

Right now, what I'm thinking about is a reasonable coverage across the timezones for each one of the external machines, and a few specific owners who should be able to debug issues, reboot the machines if needed, and even tell the Architecture Committee to mark that job as non-required for some time in the worst case scenario.

I'd like to hear from the @kata-containers/architecture-committee members, and also from the owners of the external machines that are currently plugged in our GitHub (@arronwy / @fidencio, for the TDX one; @UnmeshDeodhar / @ryansavino / @larrydewey for the AMD ones, @BbolroC / @stevenhorsman for the SE ones that will come).

**Comment by stevenhorsman** [comment]
I think this is fair. Can we just review/extend the [always green CI guidelines](https://github.com/kata-containers/ci/blob/main/Always_green_CI.md) and potentially add some extra requirement for external machines e.g.
- What it takes to make a job required - this is probably what you've mentioned above - timezone coverage with decent response time and the ability to share logs and debug info for those machines that are behind the firewall
- What it takes to make a job become not-required - instability in jobs, or not being about to keep up with the initial requirements.

**Comment by UnmeshDeodhar** [comment]
This looks good to me too. Just one more thing that could be beneficial. Once we identify all the owners, we can add their email addresses in the baseline GHA so that they will get an email if the baseline job fails. You can mark me owner for the 2 AMD systems we have in the kata CI. I am in US Central timezone. (GMT -5)

**Comment by fidencio** [comment]
> * What it takes to make a job become not-required - instability in jobs, or not being about to keep up with the above requirements in a timely manner - basically slowing down/blocking progress I guess

Without trying to nitpick, but one question that always comes to my mind is 'what does "timely manner" mean?'
I sincerely think this should be well defined in order to avoid frustrations on both sides (community trying to move on with patches, and maintainers having enough time to solve issues).

**Comment by stevenhorsman** [comment]
> > * What it takes to make a job become not-required - instability in jobs, or not being about to keep up with the above requirements in a timely manner - basically slowing down/blocking progress I guess
> 
> Without trying to nitpick, but one question that always comes to my mind is 'what does "timely manner" mean?' I sincerely think this should be well defined in order to avoid frustrations on both sides (community trying to move on with patches, and maintainers having enough time to solve issues).

I completely agree that the timely manner should be defined in the documentation, I just didn't want to pick what it is right now. The green CI doc says that job is marked as non-required after two consecutive failures, so I just that minimum time for that is 24-48hrs, which maybe feels like a good starting place, but getting consensus on it is important.

**Comment by BbolroC** [comment]
It sounds good to me. You can mark me and @hbrueckner as an owner for the future SE system. We are in the CET timezone. (GMT+1). Thanks.
